src/inc/regdisp.h

```diff
@@ -396,7 +396,7 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 #elif defined(_TARGET_X86_) // _TARGET_ARM_
     for (int i = 0; i < 7; i++)
     {
-        *(&pRD->ctxPtrsOne.Esi + i) = (&pctx->Esi + i);
+        *(&pRD->ctxPtrsOne.Esi + i) = (&pRD->pCurrentContext->Esi + i);
```

Comment:
@janvorli Sorry for not being clear. Yes, you are right. ``InitializeCurrentContextForCrawlFrame`` is called after ``FillRegDisplay``, which cause this mismatch. 

It seems that your approach would be better than the alternative approach that I mentioned above.